irksome/discontinuous_galerkin_stepper.py

```diff
@@ -0,0 +1,310 @@
+    in time method
+
+    :arg F: A :class:`ufl.Form` instance describing the semi-discrete problem
+            F(t, u; v) == 0, where `u` is the unknown
+            :class:`firedrake.Function and `v` is the
+            :class:firedrake.TestFunction`.
+    :arg order: an integer indicating the order of the DG space to use
+         (with order == 0 corresponding to DG(0)-in-time)
+    :arg t: a :class:`Function` on the Real space over the same mesh as
+         `u0`.  This serves as a variable referring to the current time.
+    :arg dt: a :class:`Function` on the Real space over the same mesh as
+         `u0`.  This serves as a variable referring to the current time step.
+         The user may adjust this value between time steps.
+    :arg u0: A :class:`firedrake.Function` containing the current
+            state of the problem to be solved.
+    :arg bcs: An iterable of :class:`firedrake.DirichletBC` containing
+            the strongly-enforced boundary conditions.  Irksome will
+            manipulate these to obtain boundary conditions for each
+            stage of the method.
+    :arg basis_type: A string indicating if the standard (discontinuous)
+            Lagrange basis is to be used for the time stages or if the Bernstein
+            basis is to be used instead
+    :arg quadrature: A :class:`FIAT.QuadratureRule` indicating the quadrature
+            to be used in time, defaulting to GL with order+1 points
+    :arg solver_parameters: A :class:`dict` of solver parameters that
+            will be used in solving the algebraic problem associated
+            with each time step.
+    :arg appctx: An optional :class:`dict` containing application context.
+            This gets included with particular things that Irksome will
+            pass into the nonlinear solver so that, say, user-defined preconditioners
+            have access to it.
+    :arg nullspace: A list of tuples of the form (index, VSB) where
+            index is an index into the function space associated with
+            `u` and VSB is a :class: `firedrake.VectorSpaceBasis`
+            instance to be passed to a
+            `firedrake.MixedVectorSpaceBasis` over the larger space
+            associated with the Runge-Kutta method
+    """
+    def __init__(self, F, order, t, dt, u0, bcs=None, basis_type=None,
+                 quadrature=None,
+                 solver_parameters=None, appctx=None, nullspace=None):
+        assert order >= 0
+        self.u0 = u0
+        self.orig_bcs = bcs
+        self.t = t
+        self.dt = dt
+        self.order = order
+        if basis_type is None:
+            basis_type = "Lagrange"
+        self.basis_type = basis_type
+
+        V = u0.function_space()
+        self.num_fields = len(V)
+
+        ufc_line = ufc_simplex(1)
+
+        if basis_type == "Lagrange":
+            self.el = DiscontinuousLagrange(ufc_line, order)
```

Comment:
This gives you equispaced points, do you get better stablity/acuracy with a different choice of points? I guess accuracy does not get affected due to basis independence for the quasi-optimality of Galerkin methods.

Reply:
Remember our recent fiat paper -- your have to to *very* high order for this to matter in Galerkin methods for Poisson, I expect similar is true for Galerkin in time.  I didn't realize that `DiscontinuousLagrange` was just equispaced.  Switching to a discontinuous element of Lagrange would fix this.